group-actors/src/main/java/org/sunbird/util/ActivitySearchRequestGenerator.java

```diff
@@ -0,0 +1,136 @@
+package org.sunbird.util;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sunbird.models.ActivitySearchRequestConfig;
+import org.sunbird.models.SearchRequest;
+import org.sunbird.service.impl.SearchActivityServiceImpl;
+import org.sunbird.util.helper.PropertiesCache;
+
+public class ActivitySearchRequestGenerator {
+
+  private static Logger logger = LoggerFactory.getLogger(ActivitySearchRequestGenerator.class);
+  private static final String ACTIVITY_CONFIG_FILE = "activityConfig.json";
+
+  static Map<String, String> activityServiceSearchUrlMap = new HashMap<>();
+  static Map<String, Map<String, Object>> activityServiceSearchRequestMap = new HashMap<>();
+  static Map<String, String> activityServiceResponseFieldMap = new HashMap<>();
+  static Map<String, Map<String, String>> activityServiceHeaderMap = new HashMap<>();
+
+  public static void init() {
+    try {
+      InputStream in =
+          SearchActivityServiceImpl.class
+              .getClassLoader()
+              .getResourceAsStream(ACTIVITY_CONFIG_FILE);
+
+      loadActivityConfigMap(in);
+
+    } catch (IOException e) {
+      logger.error("Error while loading activity Config" + e);
+    }
+  }
+
+  static void loadActivityConfigMap(InputStream in) throws IOException {
+    if (null != in) {
+      JsonParser jsonParser = new JsonParser();
+      JsonObject jsonObject = (JsonObject) jsonParser.parse(new InputStreamReader(in, "UTF-8"));
+      Map<String, Object> activitiesMap = new Gson().fromJson(jsonObject, Map.class);
+      List<Map<String, Object>> activitiesConfigList =
+          (List<Map<String, Object>>) activitiesMap.get(JsonKey.ACTIVITIES);
+      if (null != activitiesConfigList && !activitiesConfigList.isEmpty()) {
+        initializeActivityContext(activitiesConfigList);
+      }
+    } else {
+      throw new IOException("Config file does not exist");
+    }
+  }
+
+  private static void initializeActivityContext(List<Map<String, Object>> activitiesConfigList) {
+    for (Map<String, Object> activityConfig : activitiesConfigList) {
+      String baseUrl = System.getenv((String) activityConfig.get(JsonKey.BASE_URL));
+      String apiUrl = System.getenv((String) activityConfig.get(JsonKey.API_URL));
+      Map<String, String> headers = (Map<String, String>) activityConfig.get(JsonKey.HEADERS);
+      String authToken = System.getenv(headers.get(JsonKey.AUTHORIZATION));
```

Comment:
@sknirmalkar89 - this code has an implicit assumption that all APIs will have an 'Authorization' request header.